### DISCUSSION TITLE
Styleguide workflow and ideas for improvement

### DISCUSSION THREAD

**Discussion by DennisNissenUX** [Budibase/budibase#10101] [Ideas]
Hey guys,
I've been using Budibase for about a year now, and as a developer with a strong UX background, I'm consistently impressed by the possibilities it offers. I've created a style guide project in Budibase for new projects within the same company, featuring a custom navbar, complex form views, tabs with tables and detail views, and card lists with multiple functionalities. All of these elements are styled and customizable to better match a client's branding. I also added some CSS Animations. When starting a new project, I import the styleguide project and copy-paste components as needed to the new sites created. I've also recreated these components in Adobe XD, which helps me plan new apps more efficiently. With this design, I can focus on connecting data sources and onboarding users.

I'd like to start a conversation about similar processes, tips, and tricks. Additionally, I'd like to share a few ideas for improving Budibase, open for discussion:

- Enable copying and pasting of components of the component tree,  between different apps and Budibase installations to promote collaboration and maintain consistency across multiple apps, particularly when using a style guide.

- Improve handling of arrays and nested data in REST queries. Maybe make array objects in the schema of a REST query accessible in Data Providers nested in the main Data Provider like relationships in the budibase DB. 

- Allow users to select Budibase themes within an app, through a dropdown or button action, giving developers control over available theme options.

- Support building custom themes and color schemes for both Budibase UI and apps.

Do you think these improvements would be helpful and achievable?

Thanks for the read and to the amazing work of the budibase team

**Comment by melohagan** [comment]
Hey @DennisNissenUX 

Thanks for your support and feedback! There's good ideas there.

Whilst not ideal, you could have a look at this answer to see how to implement user theming: https://github.com/Budibase/budibase/discussions/7501
